### PR TITLE
rustdoc: Update document title when displaying search results

### LINKED ISSUE #26360: rustdoc: Searching should change the document title
When you search, the document title remains the same as the last page you were on. This is quite confusing if you look through your tab's browsing history. For example, you might see something like:

* (newest)
* std::cmp::Eq - Rust
* std::borrow::Borrow - Rust
* std::borrow::Borrow - Rust
* (oldest)

However, the second entry takes you to the page "Results for eq". The title of that page should also be "Results for eq".

### PR DESCRIPTION
**Pull request by jooert**
Fixes #26360.

r? @steveklabnik

### CODE REVIEW

File: src/librustdoc/html/static/main.js [hunk 1]
```diff
@@ -571,6 +571,9 @@
                 return;
             }
 
+	    // Update document title to maintain a meaningful browser history
+	    $(document).prop("title", "Results for " + query.query + " - Rust");
```
Comment (waddlesplash) [inline src/librustdoc/html/static/main.js]: You've got a tab here and the code below and above is using only spaces...

**Comment by jooert** [comment]
Promoted tabs to spaces.

**Comment by killercup** [comment]
@jooert Does this change the title back when the user exits the search mode (by pressing ESC)? (Should it?)

**Comment by jooert** [comment]
Currently, it doesn't. But we also do not add a new entry to the browser history on pressing ESC at the moment. But, although the search query is still visible in the input field after pressing ESC, I wasn't able to go back to the results for this query (other than modifying the search query), so this whole ESC-pressing feature probably needs more work anyway. (?)

**Pull request merged**